include/edgex/edgex-base.h

```diff
@@ -28,6 +28,12 @@ typedef struct devsdk_nvpairs
   struct devsdk_nvpairs *next;
 } devsdk_nvpairs;
 
+typedef struct devsdk_new_device
```

Comment:
Instead of defining a new structure to hold a list of device names, use a string map:
`iot_data_t *devices;`
Initialisation (create empty map):
`devices = iot_data_alloc_map (IOT_DATA_STRING);`
Add entry:
`iot_data_map_add (devices, iot_data_alloc_string (name, IOT_DATA_COPY), iot_data_alloc_null ());`
Test for existing name:
`if (iot_data_string_map_get (devices, name))`
Clean up
`iot_data_free (devices);`

Also, store the map as a member of the devmap or service structs rather than a global

Reply:
Thanks @iain-anderson for this suggestion... This shall be more efficient to be used...